node/services/service.go

```diff
@@ -19,8 +24,9 @@ func Block(lc fx.Lifecycle, fetcher block.Fetcher, store ipld.DAGService) *block
 }
 
 // Share constructs new share.Service.
-func Share(lc fx.Lifecycle, dag ipld.DAGService) share.Service {
-	service := share.NewService(dag)
+func Share(ctx context.Context, lc fx.Lifecycle, dag ipld.DAGService) share.Service {
+	avail := share.NewLightAvailability(merkledag.NewSession(fxutil.WithLifecycle(ctx, lc), dag))
```

Comment:
Wondering if we should not directly use the `fxutil.WithLifecycle(ctx, lc)` as a param instead?